### PR TITLE
Tests: unit tests for the native integrations

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [X] Refactoring

## :scroll: Description
unit tests for the native integration

## :bulb: Motivation and Context
we'd like to have it unit testable

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: dart/test/sentry_test.dart [hunk 1]
```diff
@@ -102,8 +102,8 @@ void main() {
     });
   });
 
-  test("options can't be null", () {
-    expect(() async => await Sentry.init((options) => options = null),
+  test("options can't be null", () async {
+    await expect(() async => await Sentry.init((options) => options = null),
```
Comment (rxlabz) [inline dart/test/sentry_test.dart]: just curious : what is the behavior without `await`( `await expect()`  ) ?

File: flutter/lib/default_integrations.dart [hunk 2]
```diff
@@ -8,32 +8,45 @@ import 'package:sentry/sentry.dart';
...
     },
   );
+}
 
-  Isolate.current.addErrorListener(receivePort.sendPort);
+/// Parse and raise an event out of the Isolate error.
+/// Visible for testing.
+Future<void> handleIsolateError(
+    Hub hub, SentryOptions options, dynamic error) async {
+  options.logger(SentryLevel.debug, 'Capture from IsolateError $error');
```
Comment (rxlabz) [inline flutter/lib/default_integrations.dart]:
```suggestion
Future<void> handleIsolateError(
    Hub hub, SentryOptions options,
    dynamic error,
) async {
  options.logger(SentryLevel.debug, 'Capture from IsolateError $error');
```

File: dart/test/sentry_test.dart [hunk 3]
```diff
@@ -102,7 +102,7 @@ void main() {
     });
   });
 
-  test("options can't be null", () {
+  test("options can't be null", () async {
```
Comment (rxlabz) [inline dart/test/sentry_test.dart]: is async still needed ?

File: flutter/test/file_system_transport_test.dart [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sentry_flutter/file_system_transport.dart';
+
+void main() {
+  const MethodChannel _channel = MethodChannel('sentry_flutter');
+
+  const _jsonDecoder = JsonDecoder();
+  const _jsonEncoder = JsonEncoder();
```
Comment (rxlabz) [inline flutter/test/file_system_transport_test.dart]: just curious : why not use `jsonEncode(...)` & `jsonDecode(...)`

File: flutter/test/sentry_flutter_test.dart [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
+import 'package:package_info/package_info.dart';
 import 'package:sentry/sentry.dart';
-import 'package:sentry_flutter/default_integrations.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+import 'sentry_flutter_util_test.dart';
 
 void main() {
-  const MethodChannel channel = MethodChannel('sentry_flutter');
+  const MethodChannel _channel = MethodChannel('sentry_flutter');
```
Comment (rxlabz) [inline flutter/test/sentry_flutter_test.dart:10]: why did you renamed it ? it's not a field, it should not have to be "private"

File: flutter/test/sentry_flutter_util.dart [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sentry_flutter/default_integrations.dart';
+import 'package:sentry_flutter/file_system_transport.dart';
+import 'package:sentry_flutter/version.dart';
+
+import 'mocks.dart';
+
+FutureOr<void> configurationTester(SentryOptions options,
+    {bool isWeb = false}) async {
```
Comment (rxlabz) [inline flutter/test/sentry_flutter_util.dart]:
```suggestion
FutureOr<void> configurationTester(
  SentryOptions options, {
  bool isWeb = false,
}) async {
```

**Review by rxlabz — APPROVED** [review]
few minor suggestions but LGTM

File: flutter/test/file_system_transport_test.dart (hunk 4 above)
Comment (marandaneto) [reply flutter/test/file_system_transport_test.dart]: I didnt know about them, thanks :)

File: dart/test/sentry_test.dart (hunk 3 above)
Comment (marandaneto) [reply dart/test/sentry_test.dart]: guess not, not sure why I've added either, removed it.

File: flutter/test/sentry_flutter_test.dart (hunk 5 above)
Comment (marandaneto) [reply flutter/test/sentry_flutter_test.dart:10]: just for the sake of encapsulation, it was not really needed

**Pull request merged**